bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -846,22 +877,42 @@ public void stop() {
 	webView.Stop();
 }
 
+private void writeToDefaultPathDOM() {
+	if(html != null && URI.create(getUrl()).equals(Browser.BASE_URI)) {
+		boolean test = jsEnabled;
+		jsEnabled = true;
+		execute("document.open(); document.write(`" + html + "`); document.close();");
+		jsEnabled = test;
+		this.html = null;
+	}
+}
+
 @Override
 public boolean setText(String html, boolean trusted) {
-	char[] data = new char[html.length() + 1];
-	html.getChars(0, html.length(), data, 0);
-	return webView.NavigateToString(data) == COM.S_OK;
+	return setWebpageData(Browser.BASE_URI.toASCIIString(), null, null, html);
 }
 
-@Override
-public boolean setUrl(String url, String postData, String[] headers) {
+private boolean setWebpageData(final String url, String postData, String[] headers, String html) {
+	// If the create() method hasn't finished executing, queue the call to wait for the browser to finish initializing
+	if (!initializationProcess.isDone()) {
+		browser.getDisplay().asyncExec(() -> {
+			waitForInitialization();
```

Comment:
If this fails (timeout or other reasons) then the code below still executes. You should contemplate this and skip the execution accordingly.
My suggestion: let `waitForInitialization()` return `boolean` and react accordingly

Reply:
waitForInitialialization would now throw an error so the following code won't execute.

Reply:
Perfect